File: src/Enterprise/C#/patterns/TemplateMethod.cs
using System;
namespace Genkidama.PatternExamples;
public static class TemplateMethodExample { public static bool Run(){string Pipe(string r,Func<string>t)=>$"{r}>{t()}>publish";return Pipe("read-csv",()=>"normalize")=="read-csv>normalize>publish";} }

File: src/Enterprise/VB.NET/patterns/TemplateMethod.vb
Imports System
Friend Module TemplateMethodExample
    Friend Function Run() As Boolean
        Dim pipeline As Func(Of String,Func(Of String),String)=Function(read,transform) $"{read}>{transform()}>publish"
        Return pipeline("read-csv",Function() "normalize")="read-csv>normalize>publish"
    End Function
End Module
